modules/notifications/notifications.php

```diff
@@ -360,25 +360,31 @@ function notifications_meta_box() {
 	 */
 	function ajax_save_post_subscriptions() {
 		global $edit_flow;
+
+		// Verify nonce.
+		if ( ! isset( $_POST['_nonce'] ) || ! wp_verify_nonce( $_POST['_nonce'], 'save_user_usergroups' ) ) {
+			die( __( 'Nonce check failed. Please ensure you can add users or user groups to a post.', 'edit-flow' ) );
+		}
+
+		$post_id = isset( $_POST['post_id'] ) ? (int) $_POST['post_id'] : 0;
+		$post    = get_post( $post_id );
+
+		$valid_post = ! is_null( $post ) && ! wp_is_post_revision( $post_id ) && ! wp_is_post_autosave( $post_id );
+		if ( ! isset( $_POST['ef_notifications_name'] ) || ! $valid_post || ! current_user_can( $this->edit_post_subscriptions_cap ) ) {
+			die();
+		}
+
+		$user_group_ids = isset( $_POST['user_group_ids'] ) && is_array( $_POST['user_group_ids'] ) ? array_map( 'intval', $_POST['user_group_ids'] ) : array();
```

Comment:
We might want to break this line up or to add a comment as it's a bit long. We want to gravitate towards verbose easy to read code vs short code. The benchmark I try to use is, could I read this code if it was 2 am and something was broken that I had to fix. So in practice that means not using the ternary operator for anything that's not a simple check (Line 369 is a good usage of it). Of note, this is my personal preference and others might have more input on this :).

Reply:
I agree, it does err on the side of "too much" for one line / a ternary. Was the result of combining the previous two lines.